crates/pipeline_manager/src/db/test.rs

```diff
@@ -1416,11 +1411,11 @@ impl Storage for Mutex<DbModel> {
         let mut hasher = sha::Sha256::new();
         hasher.update(key.as_bytes());
         let hash = openssl::base64::encode_block(&hasher.finish());
-        if s.api_keys.contains_key(&hash) {
-            Err(anyhow::anyhow!(DBError::DuplicateKey))
-        } else {
-            s.api_keys.insert(hash, permissions);
+        if let std::collections::btree_map::Entry::Vacant(e) = s.api_keys.entry(hash) {
```

Comment:
We try to avoid fully qualified pathd. Please import `std::colletions::btree_map` at the top of the module.

Reply:
I assume clippy suggested this. What's wrong with contains_key()?

Reply:
it does two lookups instead of one

Reply:
Pushed a fix.